addons/mail/static/src/messaging/service/messaging_service/messaging_service.js

```diff
@@ -18,6 +19,10 @@ const MessagingService = AbstractService.extend({
     start() {
         this._super(...arguments);
 
+        if (!('shouldRaiseEntityDeletedError' in this.env)) {
+            this.env.shouldRaiseEntityDeletedError = false;
```

Comment:
I would put this as part of the service, not in the `env`, similar to `messagingCreatedPromise`.